simulator/simulator.cpp

```diff
@@ -64,6 +63,9 @@ int main(int argc, char* argv[]) {
         } else if (strcmp(argv[i], "--smallfield") == 0) {
             Field_Dimensions::Current_Dimensions =
                 Field_Dimensions::Single_Field_Dimensions * scaling;
+        } else if (strcmp(argv[i], "--timeout") == 0) {
+            timeoutsimulator = RJ::Seconds(stoi(argv[i + 1]));
```

Comment:
You will need to do a check here to see if `argv[i+1]` is out of range or not. 

Right now this will cause the simulator to segfault if you run it like this: `./run/simulator --timeout` with no args after timeout.